kasa/smartcam/modules/camera.py

```diff
@@ -73,17 +88,40 @@ def stream_rtsp_url(self, credentials: Credentials | None = None) -> str | None:
         :return: rtsp url with escaped credentials or None if no credentials or
             camera is off.
         """
-        if not self.is_on:
-            return None
-        dev = self._device
-        if not credentials:
-            credentials = self._get_credentials()
+        if un_pw := self._get_stream_creds(credentials):
+            username, password = un_pw
 
-        if not credentials or not credentials.username or not credentials.password:
-            return None
-        username = quote_plus(credentials.username)
-        password = quote_plus(credentials.password)
-        return f"rtsp://{username}:{password}@{dev.host}:{LOCAL_STREAMING_PORT}/stream1"
+            return f"rtsp://{username}:{password}@{self._device.host}:{LOCAL_STREAMING_PORT}/stream1"
+        return None
+
+    def stream_rtsp_alt_url(self, credentials: Credentials | None = None) -> str | None:
```

Comment:
Instead of having another method, maybe we can figure out what different streams exist and use maybe an enum (or just simple string) to select, defaulting to stream1?

Reply:
Alternatively, maybe we should provide a dict of stream URLs to use? 
```
{"HD": "rtsp://foo",
 "SD": "rtsp://bar",
}
```
I'm wondering if some component that we are not querying could deliver the information about qualities etc.?

Reply:
Done